Sources/UI/Chat/Cells/MessageTextEnrichment.swift

```diff
@@ -219,9 +223,27 @@ private extension MessageTextEnrichment {
         let attributedString = self.attributedString ?? defaultAttributedString
         
         for detectedURL in detectedURLs {
-            attributedString.addAttributes([.foregroundColor: style.replyColor], range: detectedURL.range)
+            attributedString.addAttributes([.foregroundColor: style.replyColor,
+                                            .attachment: detectedURL.url],
+                                           range: detectedURL.range)
         }
         
         self.attributedString = attributedString
     }
 }
+
+// MARK: Cache
+private var cache = NSCache<NSString, NSAttributedString>()
+extension MessageTextEnrichment {
+    private var cacheKey: NSString {
+        return text as NSString
+    }
+
+    func cachedEnrich() -> NSAttributedString? {
```

Comment:
Can be just `cached()` or `cachedEnrichment()`
Or, these functions could be static and could use `MessageTextEnrichment.init` arguments for keys
Instead of
```swift
messageTextEnrichment = MessageTextEnrichment(message, style: style, enrichURLs: enrichURLs)

if let cached = messageTextEnrichment?.cachedEnrich() {
       self.messageLabel.attributedText = cached
}
```
we can do:
```swift
if let cached = MessageTextEnrichment.cached(message, style: style, enrichURLs: enrichURLs) {
  ...
}
```